cli_tools/common/utils/logging/firelog_logger.go

```diff
@@ -0,0 +1,311 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package logging
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"github.com/google/uuid"
+	"github.com/minio/highwayhash"
+)
+
+var (
+	httpClient            httpClientInterface = &http.Client{Timeout: 5 * time.Second}
+	serverURL                                 = serverURLProd
+	serverLogEnabled                          = false
+	logMutex                                  = sync.Mutex{}
+	nextRequestWaitMillis int64
+)
+
+// constants used by logging
+const (
+	serverURLProd        = "https://firebaselogging-pa.googleapis.com/v1/firelog/legacy/log"
+	key                  = "Ix3TgeJBsH2TH8FUg9_ukW6I6U0t1zOmCySazIA"
+	ImageImportAction    = "ImageImport"
+	ImageExportAction    = "ImageExport"
+	InstanceImportAction = "InstanceImport"
+	targetSizeGb         = "target-size-gb"
+	sourceSizeGb         = "source-size-gb"
+	statusStart          = "Start"
+	statusSuccess        = "Success"
+	statusFailure        = "Failure"
+)
+
+type logResult string
+
+const (
+	failedOnCreateRequest          logResult = "FailedOnCreateRequest"
+	failedOnCreateRequestJSON      logResult = "FailedOnCreateRequestJSON"
+	failedToParseResponse          logResult = "FailedToParseResponse"
+	failedAfterRetry               logResult = "FailedAfterRetry"
+	failedOnUndefinedResponse      logResult = "FailedOnUndefinedResponse"
+	failedOnMissingResponseDetails logResult = "FailedOnMissingResponseDetails"
+	serverLogDisabled              logResult = "ServerLogDisabled"
+)
+
+type httpClientInterface interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func reverse(str string) string {
```

Comment:
Move this to e.g. cli_tools/common/utils/string/string_utils.go

Reply:
It's not useful anymore, deleting